app/src/main/java/dev/baseio/discordjetpackcompose/ui/components/CountryPicker.kt

```diff
@@ -0,0 +1,189 @@
+import androidx.compose.ui.unit.dp
+import dev.baseio.discordjetpackcompose.R
+import dev.baseio.discordjetpackcompose.models.Country
+import dev.baseio.discordjetpackcompose.ui.theme.DiscordJetpackComposeTheme
+import dev.baseio.discordjetpackcompose.ui.utils.simpleVerticalScrollbar
+import dev.baseio.discordjetpackcompose.utils.Constants
+import dev.baseio.discordjetpackcompose.utils.readAssetFile
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun CountryPicker(
+    bottomSheetScaffoldState: ModalBottomSheetState,
+    backgroundContent: @Composable () -> Unit,
+    onCountrySelected: (Country) -> Unit,
+    countryList: List<Country>? = LocalContext.current.readAssetFile(Constants.CountriesAssetFilePath)
+) {
+    val coroutineScope = rememberCoroutineScope()
+    BackHandler(enabled = bottomSheetScaffoldState.isVisible) {
+        coroutineScope.launch { bottomSheetScaffoldState.hide() }
+    }
+    ModalBottomSheetLayout(
+        sheetState = bottomSheetScaffoldState, sheetContent = {
+            Box(modifier = Modifier.fillMaxSize()) {
+                var searchQuery by remember { mutableStateOf("") }
+                Column {
+                    SearchBox(
+                        currentValue = searchQuery,
+                        onValueChanged = { updatedQuery: String -> searchQuery = updatedQuery },
+                    )
+                    val lazyListState = rememberLazyListState()
+                    LazyColumn(
+                        state = lazyListState,
+                        modifier = Modifier.simpleVerticalScrollbar(
+                            state = lazyListState,
+                            color = MaterialTheme.colors.onSurface.copy(alpha = 0.5f)
+                        ),
+                    ) {
+                        countryList?.filter { country ->
+                            country.name.contains(
+                                searchQuery, ignoreCase = true
+                            )
+                        }?.let { nnCountryList ->
+                            items(nnCountryList.size) { index ->
+                                CountryItem(
+                                    countryName = nnCountryList[index].name,
+                                    countryCode = nnCountryList[index].phoneCountryCode,
+                                    onCountrySelected = { onCountrySelected(nnCountryList[index]) },
+                                )
+                            }
+                        }
+                    }
+                }
+            }
+        }, sheetShape = RoundedCornerShape(0), scrimColor = Color.Black.copy(alpha = 0.32f)
+    ) {
+        backgroundContent()
+    }
+}
+
+@Composable
+private fun CountryItem(
+    countryName: String,
+    countryCode: String,
+    onCountrySelected: () -> Unit,
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable(
```

Comment:
@shubhamsinghshubham777 can we have a global extension of this modifier which is called clickableWithRipple ?